schema/constraint_literal_value.go

```diff
@@ -39,10 +44,171 @@ func (lv LiteralValue) EmptyCompletionData(nextPlaceholder int, nestingLevel int
+			value = formatNumberVal(lv.Value)
+		}
+
+		return &HoverData{
+			Content: lang.Markdown(value),
+		}
+	}
+	if lv.Value.Type().IsListType() {
+		vals := lv.Value.AsValueSlice()
+		elemData := make([]string, len(vals))
+		for i, val := range vals {
+			c := LiteralValue{
+				Value: val,
+			}
+			hoverData := c.EmptyHoverData(nestingLevel)
+			if hoverData == nil {
+				return nil
+			}
+			elemData[i] = hoverData.Content.Value
+		}
+
+		return &HoverData{
+			Content: lang.Markdown(fmt.Sprintf(`tolist([%s])`, strings.Join(elemData, ", "))),
+		}
+	}
+	if lv.Value.Type().IsSetType() {
+		vals := lv.Value.AsValueSlice()
+		elemData := make([]string, len(vals))
+		for i, val := range vals {
+			c := LiteralValue{
+				Value: val,
+			}
+			hoverData := c.EmptyHoverData(nestingLevel)
+			if hoverData == nil {
+				return nil
+			}
+			elemData[i] = hoverData.Content.Value
+		}
+
+		return &HoverData{
+			Content: lang.Markdown(fmt.Sprintf(`toset([%s])`, strings.Join(elemData, ", "))),
+		}
+	}
+	if lv.Value.Type().IsTupleType() {
+		vals := lv.Value.AsValueSlice()
+		elemData := make([]string, len(vals))
+		for i, val := range vals {
+			c := LiteralValue{
+				Value: val,
+			}
+			hoverData := c.EmptyHoverData(nestingLevel)
+			if hoverData == nil {
+				return nil
+			}
+			elemData[i] = hoverData.Content.Value
+		}
+
+		return &HoverData{
+			Content: lang.Markdown(fmt.Sprintf(`[%s]`, strings.Join(elemData, ", "))),
+		}
+	}
+	if lv.Value.Type().IsMapType() {
+		valueMap := lv.Value.AsValueMap()
+
+		attrNames := sortedValueMap(valueMap)
+
+		data := ""
+		if nestingLevel == 0 {
+			data += "```\n"
+		}
+
+		data += "tomap({\n"
+		for _, name := range attrNames {
+			val := valueMap[name]
+
+			cons := LiteralValue{
+				Value: val,
+			}
+
+			hoverData := cons.EmptyHoverData(nestingLevel + 1)
+			if hoverData == nil {
+				return nil
+			}
+
+			data += fmt.Sprintf("%s%q = %s\n",
+				strings.Repeat("  ", nestingLevel+1),
+				name, hoverData.Content.Value)
+		}
+		data += fmt.Sprintf("%s})", strings.Repeat("  ", nestingLevel))
+		if nestingLevel == 0 {
+			data += "\n```\n"
+		}
+
+		return &HoverData{
+			Content: lang.Markdown(data),
+		}
+	}
+	if lv.Value.Type().IsObjectType() {
+		valueMap := lv.Value.AsValueMap()
+		attrs := make(ObjectAttributes, 0)
+		for name, attrValue := range valueMap {
+			aSchema := &AttributeSchema{
+				Constraint: LiteralValue{
+					Value: attrValue,
+				},
+			}
+			if lv.Value.Type().AttributeOptional(name) {
+				aSchema.IsOptional = true
+			} else {
+				aSchema.IsRequired = true
+			}
+			attrs[name] = aSchema
+		}
+		cons := Object{
+			Attributes: attrs,
+		}
+		return cons.EmptyHoverData(nestingLevel)
+	}
+
 	return nil
 }
 
+func sortedValueMap(valueMap map[string]cty.Value) []string {
+	if len(valueMap) == 0 {
+		return []string{}
+	}
+
+	constraints := valueMap
+	names := make([]string, len(constraints))
+	i := 0
+	for name := range constraints {
+		names[i] = name
+		i++
+	}
```

Comment:
```suggestion
	names := make([]string, 0, len(constraints))
	for name := range constraints {
		names = append(names, name)
	}
```

Nit: I think using `append` here makes it more readable.